posts/2020/08/versioned-hdf5-performance.md

```diff
@@ -0,0 +1,138 @@
+
+In order to analyze this, we will consider test files created with a variable number of versions (or transactions) of a dataset consisting of three ndarrays of variable length. One test includes a two-dimensional ndarray as part of this dataset, but all other test cases consist of three one-dimensional ndarrays per dataset.
+
+With each new version a certain number of rows is added, removed, and modified. For
+these tests, all changes are made to elements chosen according to a [power law](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.powerlaw.html) which biases the modifications towards the end of the array, simulating a possible use case of modifying more recent results in a given timeseries.
+
+The tests are as follows:
+
+1. A large fraction of changes is made to the dataset with each new version: The dataset initially has three arrays with 5000 rows, and 1000 positions are chosen at random and changed, and a small number (at most 10) rows are added or deleted with each new version. We will refer to this test as `test_large_fraction_changes_sparse`.
+
+2. A small fraction of changes is made to the dataset with each new version: The dataset initially has three arrays with 5000, but only 10 positions are chosen at random and changed, and a small number (at most 10) rows are added or deleted with each new version. We will refer to this test as `test_small_fraction_changes_sparse`.
+
+3. A large fraction of changes is made to the dataset with each version, with the same three arrays of 5000 rows defined initially, 1000 positions are chosen at random and changed, but the size of the final array remains constant (no new rows are added and no rows are deleted). We will refer to this test as `test_large_fraction_constant_sparse`.
+   
+In the next tests, the number of modifications is dominated by the number of appended rows. There are two such cases:
+
+4. In the first case, the dataset contains three one-dimensional arrays with 1000 rows initially, and 1000 rows are added with each new version. A small number (at most 10) values are chosen at random, following the power law described above, and changed or deleted. We call this test `test_mostly_appends_sparse`.
+
+5. In the second case, the dataset contains one two-dimensional array with shape `(30, 30)` and two one-dimensional arrays acting as indices to the 2d array. In this case, rows are only appended in the first axis of the two-dimensional array, and a small number of positions (at most 10) is chosen at random and changed. We call this test `test_mostly_appends_dense`.
+
+To test the performance of VersionedHDF5 files, we have chosen to compare a few different chunk sizes and compression algorithms. These values have been chosen arbitrarily, and optimal values depend on different use cases and on the nature of the datasets stored in the file.
```

Comment:
You're using "Versioned HDF5" for the library name (e.g. in blog title), so should this be typeset as code maybe? Or have a space?

Same applies to `VersionedHDF5` in other paragraphs

Reply:
Thanks for that. I made a pass and tried to be consistent and use "Versioned HDF5" for the library, and "versioned HDF5 files" for the actual resulting files. Let me know if this is still confusing, and I'll rephrase.